frontend/src/stimulus/controllers/dynamic/work-packages/progress/preview.controller.ts

```diff
@@ -29,21 +29,82 @@
  */
 
 import { Controller } from '@hotwired/stimulus';
+import { debounce, DebouncedFunc } from 'lodash';
+import Idiomorph from 'idiomorph/dist/idiomorph.cjs';
 
-export default class TouchedFieldMarkerController extends Controller {
+interface TurboBeforeFrameRenderEventDetail {
+  render:(currentElement:HTMLElement, newElement:HTMLElement) => void;
+}
+
+export default class PreviewController extends Controller {
   static targets = [
+    'form',
+    'progressInput',
     'initialValueInput',
     'touchedFieldInput',
-    'progressInput',
   ];
 
+  declare readonly progressInputTargets:HTMLInputElement[];
+  declare readonly formTarget:HTMLFormElement;
   declare readonly initialValueInputTargets:HTMLInputElement[];
   declare readonly touchedFieldInputTargets:HTMLInputElement[];
-  declare readonly progressInputTargets:HTMLInputElement[];
 
+  private debouncedPreview:DebouncedFunc<(event:Event) => void>;
+  private frameMorphRenderer:(event:CustomEvent<TurboBeforeFrameRenderEventDetail>) => void;
   private targetFieldName:string;
+  private touchedFields:Set<string>;
 
-  private markFieldAsTouched(event:{ target:HTMLInputElement }) {
+  connect() {
+    this.touchedFields = new Set();
+    this.touchedFieldInputTargets.forEach((input) => {
+      const fieldName = input.dataset.referrerField;
+      if (fieldName && input.value === 'true') {
+        this.touchedFields.add(fieldName);
+      }
+    });
+
+    this.debouncedPreview = debounce((event:Event) => { void this.preview(event); }, 100);
+    // TODO: Ideally morphing in this single controller should not be necessary.
+    // Turbo supports morphing, by adding the <turbo-frame refresh="morph"> attribute.
+    // However, it has a bug, and it doesn't morphs when reloading the frame via javascript.
+    // See https://github.com/hotwired/turbo/issues/1161 . Once the issue is solved, we can remove
+    // this code and just use <turbo-frame refresh="morph"> instead.
```

Comment:
Could it be that this is now fixed with 8.0.10? 

See:

https://github.com/hotwired/turbo/releases/tag/v8.0.10
https://github.com/hotwired/turbo/pull/1192

Reply:
Nice! I'll test that.

Reply:
It seems I can't make it work properly. I'll merge this one, and open another PR.